synapse/storage/databases/main/roommember.py

```diff
@@ -848,19 +847,21 @@ async def get_joined_users_from_state(
 
         assert state_group is not None
         with Measure(self._clock, "get_joined_users_from_state"):
-            return await self._get_joined_users_from_context(
-                room_id, state_group, state, context=state_entry
-            )
+            return await self._get_joined_users_from_context(state_group, state)
 
-    @cached(num_args=2, iterable=True, max_entries=100000)
     async def _get_joined_users_from_context(
         self,
-        room_id: str,
         state_group: Union[object, int],
         current_state_ids: StateMap[str],
-        event: Optional[EventBase] = None,
-        context: Optional["_StateCacheEntry"] = None,
```

Comment:
note to self/for posterity: `context` has always been not-None since #8070 removed the only call that doesn't provide it (https://github.com/matrix-org/synapse/pull/8070/files#diff-54aaace5758538a90a9ed3a020eb9b260139269ee33c10ee8b2caf850c8a3e1dL290)

Reply:
also: https://github.com/matrix-org/synapse/pull/13267 changed things so that it was always a `_StateCacheEntry`. Previously (since time immemorial) it could have been either an `EventContext` or a `_StateCacheEntry`. Both classes have properties like `delta_ids` and `prev_group`, so this worked, though it's a miracle we never broke it.